### ISSUE TITLE
Rustic cannot read password from file without newline

### ISSUE THREAD

**Issue by waspoza** [rustic-rs/rustic#426]
I have password file that doesn't end with 0xA. It works well with restic, but rustic throws error:
```
$ ./rustic -p ~/rpw.txt -r ~/mnt/nfs/rustic init
using no config file (./rustic.toml doesn't exist)
Error: error reading password file

Caused by:
    unexpected end of file
```
Would be nice to have same file usable by both programs.

**Comment by aawsome** [comment]
Thanks for opening this issue @waspoza !
You are right, a password file without `\n` doesn't work. This is actually a feature (extra check) of https://github.com/conradkleinespel/rpassword which I'm using to read the password.

As a workaround you could simply add a newline to your password file which would then work with restic as well as with rustic.
Or, you could put your password in the config file using `password=...`

**Comment by aawsome** [comment]
Is anyone willing to add a handling of the "unexpected end of file" error?

**Comment by waspoza** [comment]
Okay, since rustic is using a config file, ill just put the password there. Thanks!

**Comment by fa993** [comment]
> Is anyone willing to add a handling of the "unexpected end of file" error?

@aawsome I'd be happy to do this, but what would the expected error handling look like?

**Comment by Sathvik1007** [comment]
@aawsome 
I am new to open source and I searched but I couldn't find the file so can you tell me the file where the password is read .

**Comment by aawsome** [comment]
@Sathvik1007 I was talking about the config file whis is usually stored in $home/.config/rustic. See the examples folder for example config files.

**Comment by aawsome** [comment]
@fa993 oh, you are right, in case oft an error you can't simply take the password even without newline - the 'Err' doesn't conatain this information. I think the best is to open a feature request to the rpassword crate...

**Comment by fa993** [comment]
@aawsome Hey, I went through the docs at rpassword.... 
We use the `read_password_from_bufread` to read password from file.
Now this function was merged into the rpassword repo to answer this issue [https://github.com/conradkleinespel/rpassword/issues/19](url). 
This issue details that the above mentioned function was introduced so that it would be easy to test the crate by creating a mocking utility.
Also this is present in the source for rpassword

> //! Finally, in unit tests, you might want to pass a `Cursor`, which implements `BufRead`. In that
> //! case, you can use `read_password_from_bufread` and `prompt_password_from_bufread`:

Considering all of this, the only intention for that function is a testing utility, I wonder if a feature request to rpassword would be accepted, since we plan to use this function in production.

**Comment by aawsome** [comment]
@fa993 Thanks a lot for your research!

I think we can replace `read_password_from_bufread` by a simple `read_line()` followed by removing `\n` and `\r` in case the password end with these. Would you like to propose a PR?

**Comment by Sathvik1007** [comment]
ok

**Comment by fa993** [comment]
> @fa993 Thanks a lot for your research!
> 
> I think we can replace `read_password_from_bufread` by a simple `read_line()` followed by removing `\n` and `\r` in case the password end with these. Would you like to propose a PR?

I’d be happy to!

**Comment by aawsome** [comment]
is fixed by #442.
